### PR TITLE
Bug/1000 fix utxo send transaction

### PR DESCRIPTION
**Pull request by ciripel**
fixed BTC transaction. Now behavior is this:
The only possible way to have a dust UTXO error is close to wallet balance.
Now user can sign max amount to empty wallet.
If the user select a value too close to the max amount in this case signing is impossible due to this dust UTXO error. In this case the wallet will consider that the user is trying to send all the balance out so will instead sign the tx will max amount.

This way we do not have dust UTXO error anymore! Hooray!

### CODE REVIEW

File: lib/data/repositories/hd_wallet_repository.dart [hunk 1]
```diff
@@ -547,11 +550,37 @@ class HDWalletRepository extends WalletRepository with DecimalPlacesGetter {
           utxo: utxos.map((utxo) => utxo.toJson()).toList(),
         ),
       );
+      // TODO - change it when big decimal package updates.
+      if (tx.feeAmount.toBigInt() != BigInt.zero ||
+          BigDecimal.add(tx.feeAmount, cryptoAmount) == balance) {
+        return tx;
+      } else {
```
Comment (storkandstars) [inline lib/data/repositories/hd_wallet_repository.dart:557]: We do not need `else`  we can simply return the second condition and safe the indentation.

File: lib/logic/cubit/asset_send_summary/asset_send_summary_cubit.dart [hunk 2]
```diff
@@ -76,12 +76,14 @@ class AssetSendSummaryCubit extends Cubit<AssetSendSummaryState>
 
   Future<void> sendTransaction() async {
     final s = state;
+
     if (s is! SignedTransactionsLoadedSuccess) return;
 
     try {
       emitSafely(s.toAssetTransactionSentInProgress());
+      late BroadcastTransaction tx;
```
Comment (storkandstars) [inline lib/logic/cubit/asset_send_summary/asset_send_summary_cubit.dart:84]: Please, do not use late definition. It should be final. Or is there any reason behind it?

**Review by storkandstars — CHANGES_REQUESTED** [review]

File: lib/logic/cubit/asset_send_summary/asset_send_summary_cubit.dart (hunk 2 above)
Comment (ciripel) [reply lib/logic/cubit/asset_send_summary/asset_send_summary_cubit.dart]: fixed in [622f552](https://github.com/SimplioOfficial/simplio-app/pull/518/commits/622f55262dd87c99de087178ebcec3e0aae43d73)

File: lib/data/repositories/hd_wallet_repository.dart (hunk 1 above)
Comment (ciripel) [reply lib/data/repositories/hd_wallet_repository.dart]: fixed in [622f552](https://github.com/SimplioOfficial/simplio-app/pull/518/commits/622f55262dd87c99de087178ebcec3e0aae43d73)

File: lib/logic/cubit/asset_send_summary/asset_send_summary_cubit.dart (hunk 2 above)
Comment (ciripel) [reply lib/logic/cubit/asset_send_summary/asset_send_summary_cubit.dart]: Reason was that was a relic from a previous iteration.

**Review by storkandstars — APPROVED** [review]

**Pull request merged**